docs/configuration/configuration.md

```diff
@@ -101,6 +101,7 @@ An example [oauth2-proxy.cfg]({{ site.gitweb }}/contrib/oauth2-proxy.cfg.example
 | `-resource` | string | The resource that is protected (Azure AD only) | |
 | `-reverse-proxy` | bool | are we running behind a reverse proxy, controls whether headers like X-Real-Ip are accepted | false |
 | `-scope` | string | OAuth scope specification | |
+| `-session-store-compression` | bool | Compress session store with LZ4 (cookie session store only) | false |
```

Comment:
I figured this would be as useful option for all stores as well -- I scoped my work to the cookie session as a first step since that is what we are using in our environment.

I didn't have a redis session store handy to test those changes. I can give that a stab in another PR? Or this one (assuming we like the SessionEnvelope msgpack encoding wrapper to give details about the session coming in).